### PR TITLE
[Data Registry] additional registry queries to load extra cases

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
To support the deduplication workflow it is necessary to load 2 cases from a single case selection. Normally this would just be a casedb lookup in the form but since these cases may not be in the casedb we need to do an additional query to the registry to get the case details.

This PR adds the UI and backend changes to allow adding registry queries to case search configurations which translate into query datums in the form entry elements. See the spec & tests for examples.

Spec: https://docs.google.com/document/d/1FCSxVBoBG8LnmogMwei1nb6eC74eUhs7HWuJmBg6i3k/edit#heading=h.78n5ggjyypdw
Jira: https://dimagi-dev.atlassian.net/browse/USH-1247

image

## Feature Flag
DATA_REGISTRY

## Product Description
Give app builders the ability to make additional data registry queries to load unrelated case data.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Updated

### QA Plan
None

### Safety story
Well tested code except for the UI. Tested UI locally and also UI changes are quite isolated.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/sections/entries.py [hunk 1]
```diff
@@ -472,7 +472,7 @@ def get_datum_meta_module(self, module, use_filter=False):
             ))
 
             if module_offers_search(detail_module) and detail_module.search_config.data_registry:
-                datums.append(self.get_data_registry_query_datum(datum, detail_module))
+                datums.extend(self.get_data_registry_query_datum(datum, detail_module))
```
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/sections/entries.py:475]: Renaming to `get_data_registry_query_datums` would be a little clearer.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 2]
```diff
@@ -245,6 +245,15 @@ <h4 class="panel-title panel-title-nolink">{% trans "Search and Claim Options" %
...
                   </select>
+              </div>
+          </div>
+          <div class="form-group">
+              <label for="data-registry" class="{% css_label_class %} control-label">
+                  {% trans "Additional Data Registry Queries" %}
+                  <span class="hq-help-template" data-title="{% trans "Data Registry" %}"
+                        data-content="{% trans_html_attr "Load additional data into forms in this module by performing extra queries against the data registry." %}">
+              </label>
+              <div class="col-sm-9">
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:256]: This should either be `{% css_field_class %}` or the label above should be `col-sm-3`, so the two columns add up to 12 regardless of screen width.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html (hunk 2 above)
Comment (snopoke) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:256]: I was using `css_field_class` initially but it's quite squashed - I can revert to that though.

Comment (snopoke) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html]: eb3bab28fcc

File: corehq/apps/app_manager/suite_xml/sections/entries.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/suite_xml/sections/entries.py]: 9145e15cf6c

**Review by orangejenny — APPROVED** [review]

**Pull request merged**